File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        answer = []

        for ch in s :
            if ch in ['(', '[', '{'] :
                answer.append(ch)
            else :
                if answer :
                    if answer[-1] == '(' and ch == ")" :
                        answer.pop()
                    elif answer[-1] == '[' and ch == "]" :
                        answer.pop()
                    elif answer[-1] == '{' and ch == "}" :
                        answer.pop()
                    else :
                        return False
                else :
                    return False

        print(answer)
        
        if answer :
            return False
        else :
            return True

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for (const auto& ch : s) {
            if (ch == '(' || ch == '[' || ch == '{') {
                st.push(ch);
            }
            else {
                if (st.size() != 0) {
                    if (st.top() == '(' && ch == ')' || st.top() == '[' && ch == ']' || st.top() == '{' && ch == '}') {
                        st.pop();
                    }
                    else {
                        return false;
                    }
                }
                else {
                    return false;
                }
            }
        }

        if (st.size() == 0) {
            return true;
        }
        else {
            return false;
        }
    }
};